packages/editor/src/components/block-list/block.js

```diff
@@ -776,7 +774,25 @@ const applyWithDispatch = withDispatch( ( dispatch, ownProps, { select } ) => {
 	};
 } );
 
+const getBlock = memize( ( name, attributes ) => {
```

Comment:
I may be missing something but aren't we creating a memory leak where name and attributes are never clean up and stay in memory forever?

Reply:
If a memory leak exists I think one possible way to solve it would be something like:
```
	(() => {
	const getBlock = memize( ( name, attributes ) => {
	...
	}, { maxSize: 1 } );
	return withSelect( ( select, { clientId } ) => {
		const { getBlockName, getBlockAttributes } = select( 'core/editor' );
		const name = getBlockName( clientId );
		const attributes = getBlockAttributes( clientId );
		return {
			block: getBlock( name, attributes ),
		};
	} )
})(),
```

This way we would memoize only the last name attribute pair per BlockListBlock.

Reply:
The issue with this is if you have multiple blocks on the page, you'll keep invalidating them as you'll only keep one set of arguments. Unless I'm misunderstanding the `maxSize` property.

Reply:
You're right, this is a memory leak. We'll create a new cache entry for every change to a block's attributes (every keypress) which will never be cleared.

Reply:
It should be fixed now.